app/models/transaction.rb

```diff
@@ -51,4 +53,29 @@ def recalculate_amount
   def residue?
     self.transaction_type == 'Residue'
   end
+
+  def self.period(period)
+    case period
+    when "current_month"
+      where("transactions.created_at >= ?", Time.now.beginning_of_month)
+    when "last_3_months"
+      where("transactions.created_at >= ?", (Time.now - 3.months).beginning_of_day)
+    when "last_month"
+      last_month_begins = Time.now.beginning_of_month - 1.months
+      where("transactions.created_at >= ? AND transactions.created_at <= ?", last_month_begins,
```

Comment:
@Gataniel you may use [`BETWEEN`](http://www.postgresql.org/docs/9.4/static/functions-comparison.html) operator